### DISCUSSION TITLE
How to show sensitivity Label in Search Result

### DISCUSSION THREAD

**Discussion by vishalgoyal2472** [microsoft-search/pnp-modern-search#4093] [General]
Is it possible to show sensitivity label in Search results similar to what we can see in SharePoint online default search (as per below image at the last line)

<img width="536" alt="image" src="https://github.com/user-attachments/assets/ece54d35-4d36-486d-a9ff-a3f793ab2e59">

**Comment by kasperbolarsen** [comment]
Hi
Yes, it is possible to show a sensitivity label, but it is far from easy. I have checked the properties related to sensitivity labels, and I have only found one that seems useful: 
![image](https://github.com/user-attachments/assets/8287b682-d637-4f04-a11a-38ea5e1bb997)

I tried to map the crawled property SENSITIVITY to a RefinableString in order to see what it contained, but it remained empty :-(

So it it possible to show the label in both the Results web part and in the Refiner web part, but it both bases you will have to make a custom layout and switch on some GUIDS in order for it to work

**Reply by vishalgoyal2472** [reply]
I tried with InformationProtectionLabelID property but its showing me empty value .

**Comment by wobba** [comment]
We need work to load the labels from Graph and then match the id with the actual label to support this in PnP modern search. Not too hard, and I made the implementation in the OOB experience if anyone wants details :)

**Reply by vishalgoyal2472** [reply]
can you please share details how you have implemented ?

**Reply by wobba** [reply]
Retrieve the InformationProtectionLabelId guid, then call `https://graph.microsoft.com/beta/me/security/informationProtection/sensitivityLabels?$filter=isActive%20eq%20true` which is documented on the Graph API to load the labels and match up. The API hopefully moves to v1.0 at some point.
